scripts/SMCodeGen/templates/stateMachineTmpls/state.cpp.tt

```diff
@@ -24,11 +24,15 @@
 #include "utils/mmeTimerTypes.h"
 
 #include "mmeStates/[% FILENAME(State.Name) %].h"
-[%- FOREACH Event = State.Events %]
-[%- IF Event.NextState != "end_state" %]	
-#include "mmeStates/[% FILENAME(Event.NextState) %].h"
+[%- stateList = [] -%]
```

Comment:
Could you please tell new logic ? Old logic seems to be including files for all next states. I am not sure what new logic is doing.

Reply:
looks like i could see some example and understood what this code is doing.